actors/builtin/miner/miner_test.go

```diff
@@ -417,6 +413,68 @@ func TestWindowPost(t *testing.T) {
 		require.NoError(t, err)
 		assert.False(t, empty, "no post submission")
 	})
+
+	t.Run("skipped faults are penalized and adjust power adjusted", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+		st := getState(rt)
+
+		partitionSize := st.Info.WindowPoStPartitionSectors
+		_ = actor.commitAndProveSectors(rt, 4, 100, 1<<50, nil)
+
+		// Skip to end of proving period, cron adds sectors to proving set.
+		actor.advancePastProvingPeriodWithCron(rt)
+		st = getState(rt)
+		deadline := st.DeadlineInfo(rt.Epoch())
+
+		// advance to next dealine where we expect the first sectors to appear
+		rt.SetEpoch(deadline.Close + 1)
+		deadline = st.DeadlineInfo(rt.Epoch())
+
+		deadlines, err := st.LoadDeadlines(rt.AdtStore())
+		require.NoError(t, err)
+
+		firstPartIdx, _, err := miner.PartitionsForDeadline(deadlines, partitionSize, deadline.Index)
+		require.NoError(t, err)
+
+		partitionCount, _, err := miner.DeadlineCount(deadlines, partitionSize, deadline.Index)
+		require.NoError(t, err)
+
+		partitions := make([]uint64, partitionCount)
+		for i := uint64(0); i < partitionCount; i++ {
+			partitions[i] = firstPartIdx + i
+		}
+
+		partitionsSectors, err := miner.ComputePartitionsSectors(deadlines, partitionSize, deadline.Index, partitions)
+		require.NoError(t, err)
+		provenSectors, err := abi.BitFieldUnion(partitionsSectors...)
+		require.NoError(t, err)
```

Comment:
There's probably some part of the above code to break out to helpers functions that we'll use in other tests too.